src/main/java/com/google/cloud/tools/ide/login/AccountRoster.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.ide.login;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Internal class instantiated as a singleton member of {@link GoogleLoginState}. Manages a list
+ * of currently logged-in accounts.
+ *
+ * Not thread safe; {@link GoogleLoginState} must use it in a thread-safe way.
+ */
+class AccountRoster {
+
+  private Account activeAccount;
+  private List<Account> accounts = new ArrayList<>();
+
+  void clear() {
+    activeAccount = null;
+    accounts.clear();
+  }
+
+  boolean isEmpty() {
+    return accounts.isEmpty();
+  }
+
+  void addAndSetActiveAccount(Account account) {
+    Preconditions.checkNotNull(account.getEmail());
+
+    activeAccount = account;
+    accounts.add(account);
+  }
+
+  Account getActiveAccount() {
+    Preconditions.checkNotNull(activeAccount);
+    Preconditions.checkState(!accounts.isEmpty());
+
+    return activeAccount;
+  }
+
+  /**
+   * @return true if an account existed with the {@code email} and became active; false otherwise.
+   * @see GoogleLoginState#setActiveAccount
+   */
+  boolean setActiveAccount(String email) {
```

Comment:
setters don't usually return booleans. Something feels off here. Maybe String isn't the right argument type? E.g. what if the argument were an Account that could be added if it weren't already in the List?

And now that I think of it, why is this a List rather than a Set? If it's just for convenience, fine; but order of the accounts shouldn't matter.